test/src/e2e_vm_tests/test_programs/should_fail/abi_pub_fn_interface/src/main.sw

```diff
@@ -0,0 +1,5 @@
+contract;
+
+abi MyContract {
+    pub fn f(self) -> bool;
```

Comment:
Actually I would avoid `self` here and in the next example. Don't think this should be allowed. Do you mind adding to this case to this issue: https://github.com/FuelLabs/sway/issues/2397? Feel free to modify the description :)

Reply:
Oh, that makes sense, thanks for pointing it 👍